group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -464,29 +469,34 @@ public void testJoiningNonExistingStreamsGroupNoMissingTopics() {
             .build();
         assertTrue(coordinatorRecords.contains(CoordinatorStreamsRecordHelpers.newStreamsGroupMemberRecord(groupId, member)));
         assertTrue(coordinatorRecords.contains(CoordinatorStreamsRecordHelpers.newStreamsGroupTargetAssignmentEpochRecord(groupId, 1)));
-        assertTrue(coordinatorRecords.contains(
-            CoordinatorStreamsRecordHelpers.newStreamsGroupTargetAssignmentRecord(
+        assertTrue(coordinatorRecords.contains(CoordinatorStreamsRecordHelpers.newStreamsGroupTargetAssignmentRecord(
                 groupId,
                 member.memberId(),
-                Collections.emptyMap(),
+                Map.of(subtopologyId, new HashSet<>(List.of(0, 1, 2))),
                 Collections.emptyMap(),
                 Collections.emptyMap()
-            )
-        ));
+        )));
         assertTrue(coordinatorRecords.contains(
             CoordinatorStreamsRecordHelpers.newStreamsGroupTopologyRecord(
                 groupId,
                 topology
             )
         ));
+
+        StreamsGroupHeartbeatRequestData.TaskIds ownedActiveTasks = new StreamsGroupHeartbeatRequestData.TaskIds();
+        ownedActiveTasks.setSubtopologyId(subtopologyId);
+        ownedActiveTasks.setPartitions(List.of(0, 1, 2));
         StreamsGroupMember updatedMember = new org.apache.kafka.coordinator.group.streams.CurrentAssignmentBuilder(member)
             .withTargetAssignment(
                 1,
-                new org.apache.kafka.coordinator.group.streams.Assignment(Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap())
+                new org.apache.kafka.coordinator.group.streams.Assignment(Map.of(subtopologyId, Set.of(0, 1, 2)), Collections.emptyMap(), Collections.emptyMap())
             )
-            .withOwnedActiveTasks(Collections.emptyList())
+            .withOwnedActiveTasks(List.of(ownedActiveTasks))
             .withOwnedStandbyTasks(Collections.emptyList())
             .withOwnedWarmupTasks(Collections.emptyList())
+            .withCurrentActiveTaskProcessId((s, p) -> null)
+            .withCurrentStandbyTaskProcessIds((s, p) -> Collections.emptySet())
+            .withCurrentWarmupTaskProcessIds((s, p) -> Collections.emptySet())
```

Comment:
Also, here, all changes made to reflect that assignment can occur in the initial request.